keras_eval/eval.py

```diff
@@ -176,28 +175,29 @@ def save_probabilities_labels(self, id, save_path):
 
     def compute_inference_probabilities(self, concept_dictionary, probabilities):
         '''
-        Combines probabilities based on key "group" in concept_dictionary and saves the values in self.probabilities
+        Computes the probability inference based on key "group" in concept_dictionary and saves the values in self.probabilities
 
         Args:
             concept_dictionary: It is the dictionary which contains all the granular concepts and the mapping with the groups.
             probabilities: These are computed granular probabilities
 
         '''
+        self.combined_probabilities = utils.combine_probabilities(probabilities, self.combination_mode)
 
         for concept in concept_dictionary:
-
             if concept['group'] in self.group_id_dict.keys():
                 self.group_id_dict[concept['group']].append(concept['class_index'])
             else:
                 self.group_id_dict[concept['group']] = [concept['class_index']]
 
-        self.combined_probs = [[[0.0, ] * len(self.concept_labels)] * len(probabilities[0])]
-        self.combined_probs = np.array(self.combined_probs)
+        inference_probabilities = np.zeros((1, len(self.combined_probabilities), len(self.concept_labels)))
+
         for idx, concept_label in enumerate(self.concept_labels):
             column_numbers = self.group_id_dict[concept_label]
             for column_number in column_numbers:
-                self.combined_probs[0][:, idx] += probabilities[0][:, column_number]
-        self.probabilities = self.combined_probs
+                inference_probabilities[0][:, idx] += self.combined_probabilities[:, column_number]
+
+        self.probabilities = inference_probabilities
```

Comment:
Shouldn't this be (n_models, n_samples, n_labels) to be consistent with the rest?